tests/NTest_gpio.lua

```diff
@@ -0,0 +1,82 @@
+-- Walk the GPIO subsystem through its paces, using the attached I2C GPIO chip
+--
+--   Node GPIO 13 (index 7) is connected to I2C expander channel B6; node OUT
+--   Node GPIO 15 (index 8) is connected to I2C expander channel B7; node IN
+
+local N = require('NTest')("gpio")
+
+N.test('setup', function()
```

Comment:
That looks good as an init function. I had thought about adding `init_run`, `before_each_test` and the one afterwards to NTest but this makes the `init_run` and its counterpart obsolete at least.

Reply:
I wonder if a `N.preflight` variant on `N.test` might still be useful: sometimes if a test fails, there's no point in running any of the rest of them, and so we should just stop now.

Reply:
I was planning to have a collection of features which could be required in the test declaration (or the testrun) and a (configuration file)/(detection process for installed modules) which is checked before running a test.

But  preflight allows for dynamic checks.